File: iteration2/code/src/coursetracking/models/TakenCourse.java
package coursetracking.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class TakenCourse extends Course {

    @SerializedName(value = "letterGrade")
    @Expose
    private String letterGrade = null;

    public TakenCourse(Course c, String letterGrade) {
        super(c);
        this.letterGrade = letterGrade;
    }

    public TakenCourse(Course c) {
		super(c);
	}
    
    public String getLetterGrade() {
        return letterGrade;
    }

    public void setLetterGrade(String letterGrade) {
       this.letterGrade = letterGrade;
    }
}

File: iteration3/code/src/model/takencourse.py
from model.course import Course


class TakenCourse(Course):
    def __init__(self):
        self.__letterGrade: str = None

    def setLetterGrade(self, letterGrade: str):
        self.__letterGrade = letterGrade

    def getLetterGrade(self) -> str:
        return self.__letterGrade